drivers/gles2/rasterizer_scene_gles2.cpp

```diff
@@ -4006,7 +4006,7 @@ void RasterizerSceneGLES2::initialize() {
 
 	// cubemaps for shadows
 	if (storage->config.support_shadow_cubemaps) { //not going to be used
-		int max_shadow_cubemap_sampler_size = 512;
+		int max_shadow_cubemap_sampler_size = CLAMP(int(next_power_of_2(GLOBAL_GET("rendering/quality/shadow_atlas/size")) >> 2), 256, 2048);
```

Comment:
Many GLES2 (OpenGL 2.1 and OpenGL ES 2.0 devices) only support a max cubemap texture size of 1024x1024. The clamp here should either be reduced to 1024, or should use check the max cubemap texture size (which I think should already be available in rasterizer_storage).

Reply:
You can get max cubemap texture size and store it in config like it is done here
https://github.com/godotengine/godot/blob/5e434841ec358ae3fefa2356b122c9d62df94179/drivers/gles2/rasterizer_storage_gles2.cpp#L6219

And then just change ``2048`` to ``MIN(2048, config.max_cubemap_texture_size)``

Reply:
That all makes sense. Thanks for your guidance on this. I've made those changes, but now it seems like the max cubemap size is only set on startup. So when you have a lower resolution atlas and you increase its size, the filtering will be jaggy until you restart the engine. I assume that is a different issue (which is probably tied to #18621) and will not really be solved for 3.x? And there's no way to "restart" these settings?

Reply:
Yes, it is essentially the same as #18621. Before you PR the fact that the size didn't change wasn't really noticeable because it couldn't get any larger than it already was. This PR exposes that limitation because we are now using the appropriate sizes. 

Unfortunately, there is no way to "restart the settings" the shadow_cubemap array is initialized when the renderer is initialized. The best we can do is change the project setting to prompt the user to restart when it is changed.

you do so by changing:
https://github.com/godotengine/godot/blob/5e434841ec358ae3fefa2356b122c9d62df94179/servers/visual_server.cpp#L2409

to use ``GLOBAL_DEF_RST()``

Reply:
Although that may not be ideal, because the shadow atlas itself can be resized. So the restart may be misleading. It only requires a restart if the atlas is increase in size within a given range.

Reply:
I see, thanks for the info. I pushed the new changes.